### PR TITLE
fix ping with SendIpHeader on FreeBSD

### PR DESCRIPTION
**Pull request by wfurt**
Contributes to #14537

It seems like OS X expects length in host byte order while freebsd does not. 

cc: @Thefrank

### CODE REVIEW

**Review by filipnavara — APPROVED** [review]

**Comment by NN---** [comment]
> It seems like OS X expects length in host byte order while freebsd does not.

This is correct with being FreeBSD version dependent:
```
     Before FreeBSD 10.0 packets received on raw IP sockets had the ip_hl subtracted from the ip_len field.

     Before FreeBSD 11.0 packets received on raw IP sockets had the ip_len and ip_off fields converted to host byte order.  Packets written to raw
     IP sockets were expected to have ip_len and ip_off in host byte order.
```
https://www.unix.com/man-page/FreeBSD/4/ip/

P.S.
There is also OpenBSD which behaves differently :)

**Comment by wfurt** [comment]
Since this did not get any traction (yet?) I decided to push more fixes:
- I added version check we discussed with @sakno (fixes #63530)
- fix DF bit handling - FreeBSD needs different order once again
- change test buffer size for FreeBSD. 
ping utility fails with `ping: packet size too large: 198 > 56: Operation not permitted`. We surface this as `Timeout` and some tests were failing. It may be _nice_ to surface better error (perhaps other platforms too) but I did not want to make this any bigger. 


With this, all tests are passing for me:
```
  ===========================================================================================================
  /usr/home/furt/github/wfurt-runtime/artifacts/bin/System.Net.Ping.Functional.Tests/net7.0-Unix-Debug ~/github/wfurt-runtime/src/libraries/System.Net.Ping/tests/FunctionalTests
    Discovering: System.Net.Ping.Functional.Tests (method display = ClassAndMethod, method display options = None)
    Discovered:  System.Net.Ping.Functional.Tests (found 41 of 54 test cases)
    Starting:    System.Net.Ping.Functional.Tests (parallel test collections = on, max threads = 2)
    Finished:    System.Net.Ping.Functional.Tests
  === TEST EXECUTION SUMMARY ===
     System.Net.Ping.Functional.Tests  Total: 66, Errors: 0, Failed: 0, Skipped: 0, Time: 15.029s
```

and separate run as root since that is different code path
```
[root@d13 /usr/home/furt/github/wfurt-runtime/artifacts/bin/System.Net.Ping.Functional.Tests/net7.0-Unix-Debug]# ../../testhost/net7.0-FreeBSD-Debug-x64/dotnet exec --runtimeconfig System.Net.Ping.Functional.Tests.runtimeconfig.json --depsfile System.Net.Ping.Functional.Tests.deps.json xunit.console.dll System.Net.Ping.Functional.Tests.dll -xml testResults.xml -nologo -notrait category=OuterLoop -notrait category=failing -nocolor
  Discovering: System.Net.Ping.Functional.Tests (method display = ClassAndMethod, method display options = None)
  Discovered:  System.Net.Ping.Functional.Tests (found 41 of 54 test cases)
  Starting:    System.Net.Ping.Functional.Tests (parallel test collections = on, max threads = 2)
  Finished:    System.Net.Ping.Functional.Tests
=== TEST EXECUTION SUMMARY ===
   System.Net.Ping.Functional.Tests  Total: 66, Errors: 0, Failed: 0, Skipped: 0, Time: 7.058s
```

it is interesting to see the privileged tests run ~ 2x faster. (no real surprise)

**Comment by Thefrank** [comment]
Looks good!

If you don't have one handy, I can test this out on my FreeBSD 12.3 system in the next 2-3 days.

**Comment by Thefrank** [comment]
tests also pass on 12.3:
```
  Discovering: System.Net.Ping.Functional.Tests (method display = ClassAndMethod, method display options = None)
  Discovered:  System.Net.Ping.Functional.Tests (found 41 of 54 test cases)
  Starting:    System.Net.Ping.Functional.Tests (parallel test collections = on, max threads = 32)
  Finished:    System.Net.Ping.Functional.Tests
=== TEST EXECUTION SUMMARY ===
   System.Net.Ping.Functional.Tests  Total: 66, Errors: 0, Failed: 0, Skipped: 0, Time: 7.215s
```

File: src/libraries/Common/src/System/Net/NetworkInformation/UnixCommandLinePing.cs [hunk 1]
```diff
@@ -135,7 +136,7 @@ public static string ConstructCommandLine(int packetSize, int timeout, string ad
                 if (OperatingSystem.IsFreeBSD() || OperatingSystem.IsMacOS())
                 {
                     // OSX and FreeBSD use -h to set hop limit for IPv6 and -m ttl for IPv4
-                    if (ipv4)
+                    if (ipv4 || (OperatingSystem.IsFreeBSD() && Environment.OSVersion.Version.Major > 12))
```
Comment (scalablecory) [inline src/libraries/Common/src/System/Net/NetworkInformation/UnixCommandLinePing.cs:139]: the comment for this line could be clarified.

**Review by scalablecory — APPROVED** [review]
lgtm.

**Pull request merged**